Backpack-SwiftUI/CardCarousel/Classes/BPKCardCarousel.swift

```diff
@@ -0,0 +1,209 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKCardCarousel<Content: View>: View {
+    @Binding private var currentIndex: Int
+    private let cards: [Content]
+    private let onCardChange: () -> Void
```

Comment:
Thinking out loud, If I am a client and I am interested in getting notified about card changes, I would want to get the card index as well. With the current API, the `onCardChange` would be triggered, then the client would need to get the index from the `currentIndex`. I think it would be better to bundle them together. Like this: `private let onCardChange: (Int) -> Void`, and we no longer need the `currentIndex` to be a `binding`. what do you think?

Reply:
I'd say it should be the other way around. only have a binding exposed and NOT an `onCardChange`. if clients want to run logic when the current Index change, they can add an `onChange($currentIndex)` on their side! This is the approach we follow in most components!

Reply:
We will still need the binding for the consumer to be allowed to set the starting index and control it from their side 🙂. Happy to change though if it seems a better way of handling the card change.

Reply:
I will remove onCardChanged and let the client handle it